src/main/java/com/verily/genomewarp/GenomeWarpSerial.java

```diff
@@ -764,48 +789,55 @@ private static VCFHeader warpHeader(VCFHeader in, Map<String, Long> namesAndLeng
 
     // Takes the input BED and splits at non-DNA characters
     logger.log(Level.INFO, "Split DNA at non-DNA characters");
-    List<GenomeRange> dnaOnlyInputBED = null;
+    Map<String, List<GenomeRange>> dnaOnlyInputBEDPerChromosome = null;
     try {
-      if ((dnaOnlyInputBED = splitAtNonDNA(queryFasta, bedReader)) == null) {
+      if ((dnaOnlyInputBEDPerChromosome = splitAtNonDNA(queryFasta, bedReader)) == null) {
```

Comment:
Consider putting this region preprocessing behind a flag so that the prior implementation can still be used upon the merge of this PR